For reviewer

GBC-SR opened the pull request.

GBC-SR committed:
For reviever

GBC-SR committed:
feat: Поменял SearchBar на SearchTextField

GBC-SR committed:
feat: Добавил контекстное меню с 2 пунктами.
Добавил локализацию для меню создания/редактирования трекеров.

GBC-SR committed:
feat: Добавил кнопку Фильтр и её реализацию и локализацию

GBC-SR committed:
feat: Добавил Скрин тест основного экрана

GBC-SR committed:
feat: Добавил тёмную тему для главного экрана, создал вспомогательный enum для переключения на тёмную тему

GBC-SR committed:
feat: Добавил скрин тест для тёмной темы

GBC-SR committed:
feat: Интегрировал AppMetrica и создал класс для работы с ним

GBC-SR committed:
feat: Добавил AnalyticsService репорты в TrackerViewController

NaR0RoW commented on Tracker.xcodeproj/xcuserdata/semenrumancev.xcuserdatad/xcschemes/xcschememanagement.plist:
Папку `xcuserdata` стоит игнорировать, кладя в `.gitignore`. В ней хранится информация о состоянии проекта (какой файл открыт, какой таргет выбран и т.п.), брейкпоинты и схемы для текущего пользователя `Xcode`.
Про `gitignore` можно прочитать вот [тут](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files)

NaR0RoW commented on Tracker/Assets.xcassets/searchAndFilterPlaceholder.imageset/2@1x.png:
Картинки 1x лучше не добавлять. Они нужны были для iPhone без retina экранов, а последний такой - iPhone 4. Он не поддерживает минимальную версию приложения, поэтому можно смело удалять, чтобы не занимать лишнюю память

NaR0RoW commented on Tracker/SceneDelegate.swift:
```diff
@@ -20,14 +22,21 @@ final class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         
-        navController.tabBarItem = UITabBarItem(title: "Трекеры", image: UIImage(resource: .tracTabBar), tag: 0)
-        statisticsVS.tabBarItem = UITabBarItem(title: "Статистика", image: UIImage(resource: .statTabBar), tag: 1)
+        navController.tabBarItem = UITabBarItem(title: NSLocalizedString("trackers_title", comment: "Кнопка в тапбаре"),
```
Строковое представление локализации можно вынести в отдельную сущность, будет что-то вроде:
`Localizable.TabBar.title`

NaR0RoW commented on Tracker/Statistic/AnalyticsService.swift:
```diff
@@ -0,0 +1,58 @@
+    
+    private init() {}
+    
+    func report(event: String, screen: String, item: String? = nil) {
```
В целом, код читается как текст в книге, поэтому лучше вызываемые методы (`report`) объявлять после методов, которые их вызывают (`reportDeleteTap`)

NaR0RoW commented on Tracker/Trackers/Filters/FiltersViewController.swift:
```diff
@@ -0,0 +1,110 @@
+        case .completed: return NSLocalizedString("completed", comment: "Завершённые")
+        case .uncompleted: return NSLocalizedString("uncompleted", comment: "Незавершённые")
+        }
+    }
```
В однострочных выражениях `return` можно опускать

NaR0RoW commented on Tracker/Trackers/TrackerCellConfig/TrackerCollectionViewCell.swift:
```diff
@@ -31,7 +31,11 @@ final class TrackerCollectionViewCell: UICollectionViewCell {
+        emojiLabel.backgroundColor = UIColor { traitCollection in
+            traitCollection.userInterfaceStyle == .dark ?
+                UIColor.white.withAlphaComponent(0.3) :
+                UIColor.white.withAlphaComponent(0.3)
```
Два одинаковых условия

NaR0RoW commented on TrackerStores/TrackerStore.swift:
```diff
@@ -90,14 +90,45 @@ final class TrackerStore: NSObject {
+            
+            try context.save()
+            print("TrackerStore: Трекер обновлён")
+        }
```
Форматирование кода немного поехало, можно выделить это место и воспользоваться автоматическим форматированием в Xcode `control + I`

NaR0RoW commented on TrackerTests/TrackerTests.swift:
```diff
@@ -2,16 +2,41 @@
-        // Write your test here and use APIs like `#expect(...)` to check expected conditions.
+final class TrackersViewControllerSnapshotTests: XCTestCase {
+    
+    func testMainScreenSnapshot() {
```
Можно поправить предупреждение

NaR0RoW commented on TrackerTestsScreen/TrackerTestsScreen.swift:
Можно удалить файл

NaR0RoW commented on TrackerTests/TrackerTests.swift:
```diff
@@ -2,16 +2,41 @@
+               named: "main_screen_dark",
+               record: false
+           )
+       }
```
Тут тоже форматирование кода поехало

NaR0RoW commented on TrackerTests/TrackerTests.swift:
```diff
@@ -2,16 +2,41 @@
+final class TrackersViewControllerSnapshotTests: XCTestCase {
+    
+    func testMainScreenSnapshot() {
+        let VC = TrackersViewController()
```
Название констант принято начинать со строчной буквы

NaR0RoW commented:
Локализация не работает:

https://github.com/user-attachments/assets/20e67173-6539-408c-8592-eabea67f0a62

GBC-SR committed:
feat: Добавил коректное отображение тёмной темы для экрана создания трекера

GBC-SR committed:
feat: Добавил корректное отображение тёмной темы в экране выбора дней недели и в экране выбора категории

GBC-SR committed:
feat: Добавил тёмную тему для FiltersМС
fix: Поправил отображение сепаратора в экранах: Фильтры, выбор дней недели, выбор категории

GBC-SR committed:
feat: Сделал экран статистики и его локализацию
fix: Поправил отображение кнопки создать в создателе трекера
Поправил отступы

GBC-SR committed:
Feat: Доделал локализацию всех экранов кроме экрана онбординга
Fix: Исправил неправильную локализацию при первом запуске приложения
